### PR TITLE
fix sitemap

### PR DESCRIPTION
**Pull request by dtaylor7**
[DIV-1101 - fix sitemap](https://tools.hmcts.net/jira/browse/DIV-1101)

### CODE REVIEW

File: app/core/steps/Step.test.js [hunk 1]
```diff
@@ -373,11 +362,11 @@ describe(modulePath, () => {
     it('memoises the router', () => {
       expect(step.router).to.eql(step.router);
     });
-    it('throws an error if url is not implemented', () => {
+    it('returns null if url is not implemented', () => {
```
Comment (Rob-Doherty) [inline app/core/steps/Step.test.js]: will this `null` throw an error upstream? and if not, do we want it to?

File: app/steps/pay/card-payment-status/index.js [hunk 2]
```diff
@@ -128,6 +128,15 @@ module.exports = class CardPaymentStatus extends Step {
     return '/pay/card-payment-status';
   }
 
+  get nextStep() {
+    return {
```
Comment (Rob-Doherty) [inline app/steps/pay/card-payment-status/index.js]: was this code originally missed?

Comment (Rob-Doherty) [reply app/steps/pay/card-payment-status/index.js]: can you add unit test coverage for it just so it's fully covered, especially given it's a payment change

File: app/middleware/draftPetitionStoreMiddleware.js [hunk 3]
```diff
@@ -134,15 +135,13 @@ const saveSessionToDraftStoreAndReply = function(req, res, next) {
     return client
       .saveToDraftStore(authToken, session)
       .then(() => {
-        const successStatusCode = 200;
         res
-          .status(successStatusCode)
+          .status(statusCode.OK)
```
Comment (Rob-Doherty) [inline app/middleware/draftPetitionStoreMiddleware.js]: nice cleanup!

File: app/core/steps/Step.test.js (hunk 1 above)
Comment (mattyod) [reply app/core/steps/Step.test.js]: No and we don't want it to. It should be covered by tests. What you don't want is checks such as `typeof` and `instanceof` throwing uncaught errors. Which is what was happening (in the case of nextStep) for sitemap view.

File: app/steps/pay/card-payment-status/index.js (hunk 2 above)
Comment (mattyod) [reply app/steps/pay/card-payment-status/index.js]: This change was made before I picked this branch up:
https://github.com/hmcts/div-petitioner-frontend/pull/31/commits/e1f626141acab254758b692a9a1057ba26a2dd6e

But appears to have tests covering the outcomes.
Here:
https://github.com/hmcts/div-petitioner-frontend/blob/fix-sitemap/app/steps/pay/card-payment-status/index.test.js#L151
And Here:
https://github.com/hmcts/div-petitioner-frontend/blob/fix-sitemap/app/steps/pay/card-payment-status/index.test.js#L198

Comment (dtaylor7) [reply app/steps/pay/card-payment-status/index.js]: yes, this is required ( although not used ) by the site map so it knows what page comes next...

**Pull request merged**